lib/rules/_base.js

```diff
@@ -320,8 +325,20 @@ export default class {
         }
       }
 
-      if (ruleHandler && !plugin.isDisabled()) {
-        maybeRuleHandlerResult = ruleHandler.apply(plugin, arguments); // eslint-disable-line prefer-rest-params
+      if (
+        ruleHandler &&
+        plugin.config &&
+        (!plugin.isDisabled() || plugin._reportUnusedDisableDirectives)
```

Comment:
If we're going to report on unused, we need to still run the lint so we can find out if it would have failed.